test/framework/universal_cluster.go

```diff
@@ -349,6 +346,9 @@ func (c *UniversalCluster) DeployApp(opt ...AppDeploymentOption) error {
 			dataplaneResource = opts.appYaml
 		}
 
+		if opts.mesh == "" {
+			opts.mesh = "default"
```

Comment:
Not sure if it fits with the plan to never use `default` but maybe it should just error if `mesh` is empty?

Reply:
I think there is too much code in E2E tests that relies on this if :( I'd love to get rid of this but after migrating all the tests? I think this will be pretty easy to catch on review is someone is using the default mesh. I also plan to write README with rules and tips how to write tests after we have all envs